public/app/features/dashboard/utils/getPanelMenu.ts

```diff
@@ -141,7 +141,11 @@ export function getPanelMenu(
     shortcut: 'p s',
   });
 
-  if (contextSrv.hasAccessToExplore() && !(panel.plugin && panel.plugin.meta.skipDataQuery)) {
+  if (
+    contextSrv.hasAccessToExplore() &&
+    !(panel.plugin && panel.plugin.meta.skipDataQuery) &&
+    panel.datasource?.uid !== '-- Dashboard --'
```

Comment:
can you use the constant `SHARED_DASHBOARD_QUERY`?

Reply:
Sure, I've made the update. Thanks for the feedback